op-challenger/game/fault/contracts/faultdisputegame_test.go

```diff
@@ -321,6 +322,83 @@ func TestGetAllClaims(t *testing.T) {
 	}
 }
 
+func TestGetSubClaims(t *testing.T) {
+	for _, version := range versions {
+		// todo: backward and forward support
+		if version.Is("1.2.0") {
+			version := version
+			t.Run(version.version, func(t *testing.T) {
+				stubRpc, game := setupFaultDisputeGameTest(t, version)
+				claim0 := faultTypes.Claim{
+					ClaimData: faultTypes.ClaimData{
+						Value:    common.Hash{0xaa},
+						Position: faultTypes.NewPositionFromGIndex(big.NewInt(1)),
+						Bond:     big.NewInt(5),
+					},
+					CounteredBy:         common.Address{0x01},
+					Claimant:            common.Address{0x02},
+					Clock:               decodeClock(big.NewInt(1234)),
+					ContractIndex:       0,
+					ParentContractIndex: math.MaxUint32,
+				}
+				expectedClaims := []faultTypes.Claim{claim0}
+				block := rpcblock.ByNumber(42)
+				stubRpc.SetResponse(fdgAddr, methodClaimCount, block, nil, []interface{}{big.NewInt(int64(len(expectedClaims)))})
+
+				eventName := "Move"
+				fdgAbi := version.loadAbi()
+
+				var challgenIndex []interface{}
+				challgenIndex = append(challgenIndex, big.NewInt(int64(claim0.ParentContractIndex)))
+				claim := []interface{}{claim0.ClaimData.ValueBytes()}
+				address := []interface{}{claim0.Claimant}
+				query := [][]interface{}{challgenIndex, claim, address}
+				txHash := common.Hash{0xff}
+
+				query = append([][]interface{}{{fdgAbi.Events[eventName].ID}}, query...)
+
+				topics, err := abi.MakeTopics(query...)
+				var queryTopics []common.Hash
+				for _, item := range topics {
+					queryTopics = append(queryTopics, item[0])
+				}
+				require.NoError(t, err)
+				out := []coreTypes.Log{
+					{
+						Address: fdgAddr,
+						Topics:  queryTopics,
+						Data:    []byte{},
+						TxHash:  txHash,
+					},
+				}
+				stubRpc.SetFilterLogResponse(topics, fdgAddr, block, out)
+
+				contractCall := batching.NewContractCall(fdgAbi, fdgAddr, "move", claim0.ClaimData.Value, challgenIndex[0], claim0.ClaimData.Value, true)
+				inputData, err := contractCall.Pack()
+				require.NoError(t, err)
+
+				tx := coreTypes.NewTx(&coreTypes.LegacyTx{
+					Nonce:    0,
+					GasPrice: big.NewInt(11111),
+					Gas:      1111,
+					To:       &claim0.Claimant,
+					Value:    big.NewInt(111),
+					Data:     inputData,
+				})
+				require.NoError(t, err)
```

Comment:
This check seems unnecessary?

Reply:
nice catch. Done in https://github.com/ethstorage/optimism/pull/63/commits/8d03dda9a9ba5afbfa591672bbb5a80bbd5118cb